src/lib/icon/icon-registry.ts

```diff
@@ -116,16 +132,40 @@ export class MatIconRegistry {
     return this.addSvgIconInNamespace('', iconName, url);
   }
 
+  /**
+   * Registers an icon using an HTML string in the default namespace.
+   * @param iconName Name under which the icon should be registered.
+   * @param template SVG source of the icon.
+   */
+  addSvgIconTemplate(iconName: string, template: SafeHtml): this {
+    return this.addSvgIconTemplateInNamespace('', iconName, template);
+  }
+
   /**
    * Registers an icon by URL in the specified namespace.
    * @param namespace Namespace in which the icon should be registered.
    * @param iconName Name under which the icon should be registered.
    * @param url
    */
   addSvgIconInNamespace(namespace: string, iconName: string, url: SafeResourceUrl): this {
-    const key = iconKey(namespace, iconName);
-    this._svgIconConfigs.set(key, new SvgIconConfig(url));
-    return this;
+    return this._addSvgIconConfig(namespace, iconName, new SvgIconConfig(url));
+  }
+
+  /**
+   * Registers an icon using an HTML string in the specified namespace.
+   * @param namespace Namespace in which the icon should be registered.
+   * @param iconName Name under which the icon should be registered.
+   * @param template SVG source of the icon.
+   */
+  addSvgIconTemplateInNamespace(namespace: string, iconName: string, template: SafeHtml): this {
```

Comment:
@jelbourn I'm not a fan of having to have two sets of methods. Ideally the signature of `addSvgIconInNamespace` would be something like `addSvgIconInNamespace(namespace: string, iconName: string, data: SafeResourceUrl | SafeHtml)`, however there's no clean way of detecting which type of resource was used, because `SafeResourceUrl` and `SafeHtml` are interfaces.

Reply:
Yeah, would be better, but ultimately doesn't bother me that much.